### PR TITLE
Mqtt5 Listener Binding

### PR DESCRIPTION
**Pull request by xiazhvera**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/mqtt/Mqtt5Client.cpp [hunk 1]
```diff
@@ -18,71 +18,6 @@ namespace Aws
 {
     namespace Crt
     {
-
```
Comment (xiazhvera) [inline source/mqtt/Mqtt5Client.cpp:21]: Moved to Mqtt5Types.h so we can shared the ScopedRWLock between Mqtt5Client.cpp and Mqtt5Listener.cpp.

**Comment by xiazhvera** [comment]
Update the PR with unit tests.

File: include/aws/crt/mqtt/Mqtt5Client.h [hunk 2]
```diff
@@ -365,6 +367,26 @@ namespace Aws
...
+                 *  - void* out_data :  user defined out data as out-parameter
+                 */
+                using processingNativeClientHandler =
+                    std::function<bool(aws_mqtt5_client *, Allocator *, void *, void **)>;
+
+                /*
+                 * This is an evil help function to allow accessing the native client. It will call the
+                 * callback with native client and user_data, then return the result of the callback
+                 */
+                bool ProceedOnNativeClient(
```
Comment (xiazhvera) [inline include/aws/crt/mqtt/Mqtt5Client.h:384]: The function is a workaround to access the native client protected by the CPP mqtt5Client. This should no longer a thing after the new clean shutdown progress.

File: include/aws/crt/mqtt/Mqtt5Listener.h [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+            class AWS_CRT_CPP_API Mqtt5Listener final : public std::enable_shared_from_this<Mqtt5Listener>
+            {
+              public:
+                /**
+                 * Factory function for mqtt5 listener
+                 *
+                 * @param options: Mqtt5ListenerOptions
+                 * @param client: Mqtt5Client to listen to
+                 * @param allocator allocator to use
+                 * @return a new mqtt5 client
```
Comment (sbSteveK) [inline include/aws/crt/mqtt/Mqtt5Listener.h:39]: a new Mqtt5Listener?

**Comment by xiazhvera** [comment]
Close the PR as the Mqtt5Listener is redesigned. We will put up a new one later.

**Pull request closed without merging**